processing/src/main/java/io/druid/query/search/SearchQueryRunner.java

```diff
@@ -104,47 +108,63 @@ public SearchQueryRunner(Segment segment)
         dimsToSearch = dimensions;
       }
 
-      final BitmapFactory bitmapFactory = index.getBitmapFactoryForDimensions();
-
-      final ImmutableBitmap baseFilter =
-          filter == null ? null : filter.getBitmapIndex(new ColumnSelectorBitmapIndexSelector(bitmapFactory, index));
-
+      boolean useStorageAdapter = false;
       for (DimensionSpec dimension : dimsToSearch) {
         final Column column = index.getColumn(dimension.getDimension());
         if (column == null) {
           continue;
         }
-
-        final BitmapIndex bitmapIndex = column.getBitmapIndex();
-        ExtractionFn extractionFn = dimension.getExtractionFn();
-        if (extractionFn == null) {
-          extractionFn = IdentityExtractionFn.getInstance();
+        if (!column.getCapabilities().hasBitmapIndexes()) {
+          // If one of the columns lacks bitmap indexes, fallback to using StorageAdapter and do a per-row scan.
+          // Could optimize this by using bitmaps when available, full scan otherwise
```

Comment:
Changed this so that it will grab results from columns with bitmap indexes first, then move onto scanning the columns without bitmap indexes